src/main/java/com/epam/grid/engine/provider/queue/slurm/SlurmQueueProvider.java

```diff
@@ -0,0 +1,398 @@
+import java.util.HashMap;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.epam.grid.engine.utils.TextConstants.COMMA;
+
+
+/**
+ * A service class, that incorporates business logic, connected with Slurm Grid Engine queue.
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+@ConditionalOnProperty(name = "grid.engine.type", havingValue = "SLURM")
+public class SlurmQueueProvider implements QueueProvider {
+    private static final String SCONTROL_COMMAND = "scontrolCommand";
+    private static final String SINFO_COMMAND = "sinfo";
+    private static final String SCONTROL_CREATE_COMMAND = "create";
+    private static final String SCONTROL_UPDATE_COMMAND = "update";
+    private static final String SCONTROL_DELETE_COMMAND = "delete";
+
+    private static final String SCONTROL_ACTION = "command";
+    private static final String SCONTROL_USER_GROUPS = "allowedUserGroups";
+    private static final String SCONTROL_HOSTS = "hostList";
+    private static final String SCONTROL_PARTITION_NAME = "partitionName";
+    private static final String STANDARD_SLURM_DELIMETER = "\\|";
+    private static final String LEFT_SQUARE_BRACKET_SIGN = "\\[";
+    private static final String HYPHEN_SIGN = "-";
+
+    private static final Pattern NODES_RANGE_REGEX = Pattern.compile("[a-zA-Z]+\\[\\d+-\\d+]");
+    private static final int SCONTROL_OUPUT_SIZE = 42;
+    private static final int SCONTROL_OUPUT_PARTNAME_INDEX = 33;
+    private static final int SCONTROL_OUPUT_NODES_INDEX = 31;
+    private static final int SCONTROL_OUPUT_USERGROUPS_INDEX = 6;
+    private static final int SCONTROL_OUPUT_CPUS_INDEX = 2;
+
+    private final SimpleCmdExecutor simpleCmdExecutor;
+
+    /**
+     * An object that forms the structure of an executable command according to a template.
+     */
+    private final GridEngineCommandCompiler commandCompiler;
+
+    /**
+     * Returns Slurm Grid Engine.
+     *
+     * @return current engine type - Slurm Grid Engine
+     */
+    @Override
+    public CommandType getProviderType() {
+        return CommandType.SLURM;
+    }
+
+    @Override
+    public Queue registerQueue(final QueueVO registrationRequest) {
+        checkRegistrationRequest(registrationRequest);
+        final Context context = prepareContext(registrationRequest);
+        final CommandResult result = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SCONTROL_COMMAND, context));
+        checkIsResultIsCorrect(result);
+        return Queue.builder()
+                .name(registrationRequest.getName())
+                .hostList(registrationRequest.getHostList())
+                .allowedUserGroups(registrationRequest.getAllowedUserGroups())
+                .slots(new SlotsDescription(2, Collections.emptyMap()))
```

Comment:
I guess we can register a new queue with one and only slot (host) when we make request like 
```{
  "allowedUserGroups": [
    "users"
  ],
  "hostList": [
    "worker1"
  ],
  "name": "someName"
}
```
also we could place ("hostName", 1) entries to the `SlotsDescription.slotsDetails` field

Reply:
fixed